projects/bp-strapi/src/middlewares/growthbook-client.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * `growthbook-client` middleware
+ */
+
+import { GrowthBook, setPolyfills } from "@growthbook/growthbook";
+import { Strapi } from "@strapi/strapi";
+
+export default (_, { strapi }: { strapi: Strapi }) => {
+  setPolyfills({
+    // Required for Node 17 or earlier
+    fetch: require("cross-fetch"),
+    // Optional, can make feature rollouts faster
+    EventSource: require("eventsource"),
+  });
+
+  let currentCtx = null;
+
+  // Create a GrowthBook Context
+  const growthbook = new GrowthBook({
+    apiHost: "https://growthbook.harz-history.de/proxy",
+    clientKey: "sdk-g7fN7JoMErl22b8e",
+    enableDevMode: true,
+    trackingCallback: (experiment, result) => {
+      const response = currentCtx.koaContext.response;
+      let growthbookHeader = response.get("x-growthbook");
+      growthbookHeader ||= "[]";
+      const growthbookEvents = JSON.parse(growthbookHeader);
+      growthbookEvents.push({
+        experimentId: experiment.key,
+        variationId: "v" + String(result.variationId),
+      });
+      response.set("x-growthbook", JSON.stringify(growthbookEvents));
+      response.append("Access-Control-Expose-Headers", "x-growthbook");
+    },
+  });
+  // Add your own logic here.
+  return async (ctx, next) => {
+    ctx.state.withGrowthBook = (theCtx, f) => {
```

Comment:
is this called anywhere? if not, how does the `trackingCallback get its `currentCtx`?

Reply:
This PR just lays the foundational API for future work with experiments. That's why `withGrowthBook` is called nowhere (yet).